templates/web/base/admin/users/form.html

```diff
@@ -9,6 +9,29 @@
     <ul class="no-bullets">
         [% PROCESS 'admin/users/_form_details.html' %]
 
+        <li>
+          <div class="admin-hint">
+            <p>
+              [% loc("Mark users whose behaviour you want to keep a check on as <strong>flagged</strong>.") %]
+              <br>
+              [% tprintf(loc("Flagged users are listed on the <a href='%s'>flagged</a> page."), c.uri_for_action( 'admin/flagged' )) %]
+              <br>
+              [% loc("You can add an abusive user's email to the abuse list, which automatically hides (and never sends) reports they create.") %]
+            </p>
+          </div>
+
+          <label>
+            [% loc('Flagged:') %]
+            <input type="checkbox" id="flagged" name="flagged"[% user.flagged ? ' checked' : '' %]>
+          </label>
+          [% IF username_in_abuse %]
+           <small>[% loc('User in abuse table') %]</small>
+           <input class="btn" name="unban" type="submit" value="[% loc('Unban') %]">
+          [% ELSE %]
+           <input type="submit" class="btn" name="banuser" value="[% loc('Ban user') %]">
```

Comment:
My first thought was, shouldn’t this "Ban" button be down next to all the other "Danger" buttons, at the end of the form?

But then I thought, really, isn’t this just a state (banned / not banned), and shouldn’t it just be toggled like all the other states (flagged / not flagged, verified / not verified)?

Reply:
I think it is because it's held separately internally, and don't want every user edit to have to check the table if it's not ticked, if you see what I mean.